content/en/blog/_posts/2024-08-20-websocket-transition.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: blog
+title: 'Streaming Transitions from SPDY to WebSockets'
+date: 2024-08-20
+slug: websockets-transition
+author: >
+  [Sean Sullivan](https://github.com/seans3) (Google)
+  [Shannon Kularathna](https://github.com/shannonxtreme) (Google)
+---
+
+In Kubernetes 1.31, by default kubectl now uses the WebSocket protocol
+instead of SPDY for streaming.
+
+This post describes what these changes mean for you and why these streaming APIs
+matter.
+
+## Streaming APIs in Kubernetes
+
+In Kubernetes, specific endpoints that are exposed as an HTTP or RESTful
+interface are upgraded to streaming connections, which require a streaming
+protocol. Unlike HTTP, which is a request-response protocol, a streaming
+protocol provides a persistent connection that's bi-directional, low-latency,
+and lets you interact in real-time. Streaming protocols support reading and
+writing data between your client and the server, in both directions, over the
+same connection. This type of connection is useful, for example, when you create
+a shell in a running container from your local workstation and run commands in
+the container.
+
+## Why change the streaming protocol?
+
+Before the v1.31 release, Kubernetes used the SPDY/3.1 protocol by default when
+upgrading streaming connections. SPDY/3.1 has been deprecated for eight years,
+and it was never standardized. Many modern proxies, gateways, and load balancers
+no longer support the protocol. As a result, you might notice that commands like
+`kubectl cp`, `kubectl attach`, `kubectl exec`, and `kubectl port-forward`
+stop working when you try to access your cluster through a proxy or gateway.
+
+As of Kubernetes v1.31, SIG API Machinery have modified the streaming
```

Comment:
A minor typo:
It should be: "SIG API Machinery **has** modified the streaming..."

Reply:
`have` changed to `has`